core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/ContainerManager.scala

```diff
@@ -395,64 +466,80 @@ object ContainerManager {
+            (scheduledPair :: pairs, invokers)
           }
         } else {
           val wantedInvokers = candidates.filter(health => requiredResources.toSet.subsetOf(health.id.tags.toSet))
           if (wantedInvokers.nonEmpty) {
-            chooseInvokerFromCandidates(wantedInvokers, invokers, pairs, msg)
+            val scheduledPair = chooseInvokerFromCandidates(wantedInvokers, msg)
+            val updatedInvokers =
+              updateInvokerMemory(scheduledPair.invokerId, msg.whiskActionMetaData.limits.memory.megabytes, invokers)
+            (scheduledPair :: pairs, updatedInvokers)
           } else if (resourcesStrictPolicy) {
-            sendState(
-              FailedCreationJob(
-                msg.creationId,
-                msg.invocationNamespace,
-                msg.action,
-                msg.revision,
-                NoAvailableResourceInvokersError,
-                s"No available invokers with resources $requiredResources."))
-            (pairs, candidates)
+            val scheduledPair =
+              ScheduledPair(
+                msg,
+                invokerId = None,
+                Some(NoAvailableResourceInvokersError),
+                Some(s"No available invokers with resources $requiredResources."))
+            (scheduledPair :: pairs, invokers)
           } else {
             val (noTaggedInvokers, taggedInvokers) = candidates.partition(_.id.tags.isEmpty)
             if (noTaggedInvokers.nonEmpty) { // choose no tagged invokers first
-              chooseInvokerFromCandidates(noTaggedInvokers, invokers, pairs, msg)
+              val scheduledPair = chooseInvokerFromCandidates(noTaggedInvokers, msg)
+              val updatedInvokers =
+                updateInvokerMemory(scheduledPair.invokerId, msg.whiskActionMetaData.limits.memory.megabytes, invokers)
+              (scheduledPair :: pairs, updatedInvokers)
             } else {
               val leftInvokers =
                 taggedInvokers.filterNot(health => requiredResources.toSet.subsetOf(health.id.tags.toSet))
-              if (leftInvokers.nonEmpty)
-                chooseInvokerFromCandidates(leftInvokers, invokers, pairs, msg)
-              else {
-                sendState(
-                  FailedCreationJob(
-                    msg.creationId,
-                    msg.invocationNamespace,
-                    msg.action,
-                    msg.revision,
-                    NoAvailableInvokersError,
-                    s"No available invokers."))
-                (pairs, candidates)
+              if (leftInvokers.nonEmpty) {
+                val scheduledPair = chooseInvokerFromCandidates(leftInvokers, msg)
+                val updatedInvokers =
+                  updateInvokerMemory(
+                    scheduledPair.invokerId,
+                    msg.whiskActionMetaData.limits.memory.megabytes,
+                    invokers)
+                (scheduledPair :: pairs, updatedInvokers)
```

Comment:
While iterating all container creation messages with `foldLeft`, we need to keep updating the memory of the given invoker set when any message is scheduled.
So this would be the return value of the foldLeft, it returns the scheduled pairs with invokers with updated resources.